### ISSUE TITLE
N + 1 queries in TodoController.index

### ISSUE THREAD

**Issue by lelelelemon** [TracksApp/tracks#2087]
The attachments can be eager loaded with the done todos to avoid N + 1 queries.

**Comment by LuiGiovanni** [comment]
Hello! I'm interested in working on this issue, this is my first time contributing what is the process I have to follow? @lelelelemon

**Comment by ZeiP** [comment]
Hi @LuiGiovanni! Great that you're thinking of working on the issue. What kind of process are you looking for? To work on the issue you should first install a Tracks environment; you can find installation instructions in https://github.com/TracksApp/tracks/wiki/Installation .

Once you've made the necessary changes, make a fork of the repository using the Fork button in the topright corner, push the changes to your fork and then make a Pull request using the Pull request tab above. You can find more information on the process in GitHub documentation: https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/proposing-changes-to-your-work-with-pull-requests

If you need any more help, feel free to join our chat channel at https://matrix.to/#/#tracks:matrix.org .

**Comment by LuiGiovanni** [comment]
Hi @ZeiP! Excellent, I'm currently working on a mac and seeing the instructions it seems that the installation is either not possible for mac anymore o more complicated, is there something I may be missing?

**Comment by ZeiP** [comment]
I haven't used Mac myself, but it certainly should be possible to install Tracks on a Mac. The easiest way is probably using Docker: https://docs.docker.com/docker-for-mac/install/  but also the custom installation should be possible.

**Comment by LuiGiovanni** [comment]
Right, I think I'll go for the docker route is I can't work directly on this machine, thanks!

**Comment by LuiGiovanni** [comment]
@ZeiP Reviewing the issue itself, would it be advisable to use [bullet](https://github.com/flyerhzm/bullet) to resolve this issue, or is there something else you would recommend? I am open to any ideas

**Comment by LuiGiovanni** [comment]
@ZeiP Hello, running around with Bullet, I noticed that in the file app/models/todo.rb in the line 68 there is a DEFAULT_INCLUDES which is, from what i believe, the culprit for this issue. Removing this include from the index of todos_controller.rb seems to fix it, what use is there to this constant?

**Comment by LuiGiovanni** [comment]
Hello, sadly I will have to stop my work on this issue if anyone wishes to work on it, go for it. Thank you so much for everything!
